opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -140,7 +140,8 @@ namespace Opm
             // The variable perf loops over the number_of_perforations_ of *this* process.
             // This is why we do not need to convert perf to a local id, it is already *local*.
             const int cell_idx = this->well_cells_[perf];
-            this->cell_perforation_depth_diffs_[perf] = depth_arg[cell_idx] - this->perf_depth_[perf];
+            // Here we need to access the depth at the global perforation index though!
+            this->cell_perforation_depth_diffs_[perf] = depth_arg[cell_idx] - this->perf_depth_[this->pw_info_.localToGlobal(perf)];
```

Comment:
i think it would be a very good idea to keep variable names consistent. here perf is a local index, while in other spots local indices are explicitly local_perf_index and perf is global. (i understand how that happened, just saying..)